app/main.py

```diff
@@ -1 +1,56 @@
-# write your code here
+from typing import List
+
+
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None:
+        self.name: str = name
+        self.skills: List[str] = []
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        # Порядок как в тестах: ["JavaScript", "CSS", "HTML"]
+        self.skills.extend(["JavaScript", "CSS", "HTML"])
+
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills.extend(["Python", "SQL", "Django"])
+
+    def create_powerful_api(self) -> str:
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+
+class AndroidDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills.extend(["Java", "Android studio"])
+
+    def create_smooth_mobile_app(self) -> str:
+        print(f"{self.name} is creating a mobile app...")
+        return "Ads every three swipes"
+
+
+class FullStackDeveloper(BackendDeveloper, FrontendDeveloper):
+    def __init__(self, name: str) -> None:
+        # Инициализируем только базовый класс,
+        # а нужные навыки добавляем вручную один раз.
+        SoftwareEngineer.__init__(self, name)
+        self.skills.extend(
+            ["Python", "SQL", "Django", "JavaScript", "CSS", "HTML"],
+        )
```

Comment:
The `__init__` method is unnecessary here because we inherit the required attributes from the parent classes.